From 13de38c439943fe2a61a7787aa8ef925041fc4c3 Mon Sep 17 00:00:00 2001
Message-Id: <13de38c439943fe2a61a7787aa8ef925041fc4c3.1443021080.git.dledford@redhat.com>
In-Reply-To: <252749ced9cacc991f116ebc99d4f5e4e15e55cc.1443021080.git.dledford@redhat.com>
References: <252749ced9cacc991f116ebc99d4f5e4e15e55cc.1443021080.git.dledford@redhat.com>
From: Alexey Ishchuk <aishchuk@linux.vnet.ibm.com>
Date: Wed, 19 Nov 2014 11:17:18 +0100
Subject: [PATCH libibverbs 06/12] libibverbs: add support for the s390x
 platform
To: linux-rdma@vger.kernel.org

This patch adds the required platform specific code to allow execution of
the libibverbs functions on the s390x platform.

Signed-off-by: Alexey Ishchuk <aishchuk@linux.vnet.ibm.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 include/infiniband/arch.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/infiniband/arch.h b/include/infiniband/arch.h
index 88364ce5a2f3..bc1738aa7c5c 100644
--- a/include/infiniband/arch.h
+++ b/include/infiniband/arch.h
@@ -115,6 +115,13 @@ static inline uint64_t ntohll(uint64_t x) { return x; }
 #define wmb()	 mb()
 #define wc_wmb() wmb()
 
+#elif defined(__s390x__)
+
+#define mb()	{ asm volatile("" : : : "memory"); }	/* for s390x */
+#define rmb()	mb()					/* for s390x */
+#define wmb()	mb()					/* for s390x */
+#define wc_wmb() wmb()					/* for s390x */
+
 #else
 
 #warning No architecture specific defines found.  Using generic implementation.
-- 
2.4.3

